### PR TITLE
Allow for vaulted templates in template lookup

### LINKED ISSUE #34209: Template lookup filter should decrypt vault-encrypted templates
##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
template lookup filter

##### ANSIBLE VERSION
```
ansible 2.4.1.0
```

##### OS / ENVIRONMENT
N/A

##### SUMMARY
The template module/action plugin decrypts vault-encrypted templates before rendering them, but the template lookup filter plugin does not.

### PR DESCRIPTION
**Pull request by mkrizek**
##### SUMMARY

Fixes #34209

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

lib/ansible/plugins/lookup/template.py

### CODE REVIEW

File: lib/ansible/plugins/lookup/template.py [hunk 1]
```diff
@@ -71,43 +71,54 @@ def run(self, terms, variables, **kwargs):
...
-                    vars.update(generate_ansible_template_vars(lookupfile))
-                    vars.update(lookup_template_vars)
-                    self._templar.set_available_variables(vars)
-
-                    # do the templating
-                    res = self._templar.template(template_data, preserve_trailing_newlines=True,
-                                                 convert_data=convert_data_p, escape_backslashes=False)
-                    ret.append(res)
+                # Get vault decrypted tmp file
+                tmp_lookupfile = self._loader.get_real_file(lookupfile)
```
Comment (bcoca) [inline lib/ansible/plugins/lookup/template.py]: if you are using loader, just use `get_file_contents`

Comment (mkrizek) [reply lib/ansible/plugins/lookup/template.py]: Ah, much simpler. Fixed.

File: lib/ansible/plugins/lookup/template.py [hunk 2]
```diff
@@ -73,41 +73,41 @@ def run(self, terms, variables, **kwargs):
...
+                    # our search paths aren't actually the proper ones for jinja includes.
+                    # We want to search into the 'templates' subdir of each search path in
+                    # addition to our original search paths.
+                    newsearchpath = []
+                    for p in searchpath:
+                        newsearchpath.append(os.path.join(p, 'templates'))
+                        newsearchpath.append(p)
+                    searchpath = newsearchpath
+                else:
+                    searchpath = [self._loader._basedir, os.path.dirname(lookupfile)]
```
Comment (bcoca) [inline lib/ansible/plugins/lookup/template.py]: this looks like preexisting issue, the lookupfile's dirname should always be in searchpath

Comment (mkrizek) [reply lib/ansible/plugins/lookup/template.py]: https://github.com/ansible/ansible/pull/50488

Comment (mkrizek) [reply lib/ansible/plugins/lookup/template.py]: Resolved in ^^^ and rebased.

**Pull request merged**